File: tecnicell.client/src/app/Interfaces/business/Models/Sale.ts
import { Currency } from "./Currency";

export interface Sale {
    saleCode?:               string;
    currencyCode?:           string;
    warranty?:               Date;
    cost?:                   number;
    currencyCodeNavigation?: Currency;
}
export interface SaleViewModel {
    saleCode?:               string;
    currencyCode?:           string;
    warranty?:               Date;
    cost?:                   number;
    currencyViewModel?: Currency;
}

File: Tecnicell.Server/Models/Entity/Sale.cs
using System;
using System.Collections.Generic;

namespace Tecnicell.Server.Models.Entity;

public partial class Sale
{
    public string SaleCode { get; set; } = null!;

    public string? CurrencyCode { get; set; }

    public DateTime? Warranty { get; set; }

    public decimal? Cost { get; set; }

    public virtual ICollection<AccessoryHistory> AccessoryHistories { get; set; } = new List<AccessoryHistory>();

    public virtual ICollection<BatteryHistory> BatteryHistories { get; set; } = new List<BatteryHistory>();

    public virtual Currency? CurrencyCodeNavigation { get; set; }

    public virtual ICollection<PhoneHistory> PhoneHistories { get; set; } = new List<PhoneHistory>();

    public virtual ICollection<PhoneRepairHistory> PhoneRepairHistories { get; set; } = new List<PhoneRepairHistory>();

    public virtual ICollection<ScreenHistory> ScreenHistories { get; set; } = new List<ScreenHistory>();
    public virtual ICollection<DiaryWork> DiaryWorks { get; set; } = new List<DiaryWork>();

}
